src/components/print/PrintDirective.js

```diff
@@ -541,6 +541,19 @@ goog.require('ga_urlutils_service');
         if (!layer.visible || layer.opacity == 0) {
           return;
         }
+        // Only print layer which have an extent intersecting the print extent
+        var layerExtent = layer.getExtent();
+        if (layerExtent) {
+          var intersection = ol.extent.getIntersection(layerExtent,
```

Comment:
I guess you don't care to get the intersection extent, you can use this instead:

ol.extent.intersects(extent1, extent2)